includes/Modules/Site_Verification.php

```diff
@@ -115,72 +115,90 @@ protected function create_data_request( $method, $datapoint, array $data = array
 							);
 						};
 					}
+
 					$current_url = ! empty( $data['siteURL'] ) ? $data['siteURL'] : $this->context->get_reference_site_url();
 					$site        = new \Google_Service_SiteVerification_SiteVerificationWebResourceGettokenRequestSite();
 					$site->setIdentifier( $current_url );
 					$site->setType( 'SITE' );
 					$request = new \Google_Service_SiteVerification_SiteVerificationWebResourceGettokenRequest();
 					$request->setSite( $site );
 					$request->setVerificationMethod( 'META' );
-					$service = $this->get_service( 'siteverification' );
-					return $service->webResource->getToken( $request ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName
+
+					return $this->get_siteverification_service()->webResource->getToken( $request );
 			}
 		} elseif ( 'POST' === $method ) {
 			switch ( $datapoint ) {
-				case 'siteverification':
+				case 'verification':
 					if ( ! isset( $data['siteURL'] ) ) {
 						/* translators: %s: Missing parameter name */
 						return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'siteURL' ), array( 'status' => 400 ) );
 					}
+
 					return function() use ( $data ) {
 						$current_user = wp_get_current_user();
+
 						if ( ! $current_user || ! $current_user->exists() ) {
 							return new WP_Error( 'unknown_user', __( 'Unknown user.', 'google-site-kit' ) );
 						}
-						$site = $this->get_data( 'siteverification-list', $data );
+
+						$site = $this->get_data( 'verification', $data );
+
 						if ( is_wp_error( $site ) ) {
 							return $site;
 						}
+
 						$sites = array();
-						if ( isset( $site['verified'] ) && ! $site['verified'] ) {
-							$token = $this->get_data( 'siteverification-token', $data );
+
+						if ( ! empty( $site['verified'] ) ) {
+							return $site;
```

Comment:
We need to still call `$this->authentication->verification()->set( true );` before returning here (see current behavior).

Reply:
Good catch. Added!